### ISSUE TITLE
[BUG]: Concurrent monitoring: CPU data deletion mistiming can lead to no results.

### ISSUE THREAD

**Issue by GitTorre** [microsoft/service-fabric-observer#266]
## Bug Details

**Scope**

This issue applies to AppObserver, and only when AppObserverEnableConcurrentMonitoring is enabled, and you are monitoring _several_ services (which is the only reason you would enable concurrent monitoring on capable hardware).

**Problem**

Random occurrences of FabricResourceUsageData.Data containers not containing any items for CPU data. This is not consistent or frequent but _can_ take place due to a data clearing bug in ObserverBase.ProcessResourceDataReportHealth. 

**Solution**

Until 3.2.6 ships, if you experience this often, disable AppObserverEnableConcurrentMonitoring configuration in ApplicationManifest.xml. 

The easiest solution for you for now, if you build the code yourself, is to replace the ```data.ClearData()``` call at the end of the ObserverBase.ProcessResourceDataReportHealth function with 

``` C# 

if (ObserverName != ObserverConstants.AppObserverName) 
{
     data.ClearData();
}
```

**Comment by GitTorre** [comment]
Fix is in develop branch.

**Comment by GitTorre** [comment]
Fix Shipped in 3.2.6.
